stl/inc/flat_map

```diff
@@ -862,27 +864,30 @@ private:
         _Clear_flat_map_scope_guard _Guard{this};
 
         // Insert the new elements at the end
-        size_type _OldSize = size();
-        size_type _NewSize = _STD distance(_First, _Last);
+        const size_type _Old_size = size();
+        if constexpr (_Is_cpp17_fwd_iter_v<_InputIterator>) {
+            const auto _New_size = static_cast<size_type>(_STD distance(_First, _Last));

```

Comment:
Pre-existing-ish: I believe this is a bug. When containers (or adaptors) take arbitrary iterator ranges, they should use `_Convert_size<size_type>` instead of `static_cast<size_type>`. We could be a 32-bit container (adaptor) taking a 64-bit arbitrary iterator range (or 16 vs. 32, etc.).

We should check all occurrences of `static_cast<size_type>`.

Reply:
This occurrence is to be removed. The other occurrence of `static_cast<size_type>` in `<flat_map>` is fine as its source is the distance of `equal_range`.